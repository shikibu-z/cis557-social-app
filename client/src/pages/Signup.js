import React, { useState } from 'react';
import '../stylesheets/Signup.css';
import {
  useHistory,
} from 'react-router-dom';
import {
  Alert, Modal,
} from 'react-bootstrap';
import NavBar from '../components/NavBar';
import loginValidate from '../app/loginValidate';

import { userRegister } from '../api/fetchers';

function Signup() {
  const [alert, setAlert] = useState(false);
  const [alertMessage, setAlertMessage] = useState('');
  const [modalShow, setModalShow] = useState(false);
  const history = useHistory();

  const handleSignUp = async () => {
    // SignUpLogic get all user input
    const email = document.getElementById('create-email-input').value;
    const username = document.getElementById('create-username-input').value;
    const password = document.getElementById('create-password-input').value;
    const female = document.getElementById('flexRadioDefault1').checked;
    const male = document.getElementById('flexRadioDefault2').checked;
    const neutral = document.getElementById('flexRadioDefault3').checked;
    let gender = '';
    if (female) {
      [gender] = Object.keys({ female });
    } else if (male) {
      [gender] = Object.keys({ male });
    } else if (neutral) {
      [gender] = Object.keys({ neutral });
    }
    const userInfo = {
      email, username, password, gender,
    };

    // validate user input format
    const userObject = loginValidate(userInfo);
    if (Object.values(userObject).some((prop) => prop === null)) {
      /* eslint-disable-next-line no-restricted-syntax */
      for (const key in userObject) {
        if (userObject[key] === null) {
          switch (key) {
            case 'email':
              setAlert(true);
              setAlertMessage('Invalid user email!');
              return;
            case 'username':
              setAlert(true);
              setAlertMessage('Invalid username!');
              return;
            case 'password':
              setAlert(true);
              setAlertMessage('Invalid password!');
              return;
            default:
              // no need for anything, there has to be at least one case
              break;
          }
        }
      }
    }

    // login endpoint
    const response = await userRegister(userObject);
    if (response.status === 201) {
      // if success,
      setModalShow(true);
      history.push('/login');
    } else {
      setAlert(true);
      setAlertMessage('Email or username already exists!');
    }
  };

  return (
    <div>
      <NavBar isLogin={1} />
      <div className="container container d-flex align-items-center flex-column">
        <img id="login-box-logo" src={`${process.env.PUBLIC_URL}Connexion-logos_transparent.png`} alt="connexion" />
        <form className="signupbox container d-flex flex-column">
          {alert && (
            <Alert show={alert} variant="danger">
              {alertMessage}
            </Alert>
          )}
          <div className="form-group">
            <label htmlFor="create-email-input">
              Enter your Email address
              <input type="email" className="form-control" id="create-email-input" placeholder="Enter email" />
            </label>
          </div>
          <div className="form-group">
            <label htmlFor="create-username-input">
              Create a Username:
              <input type="text" className="form-control" id="create-username-input" placeholder="Enter username" />
            </label>
          </div>
          <div className="form-group">
            <label htmlFor="create-password-input">
              Create a Password:
              <div>
                <p style={{ 'font-size': '10px', margin: 0 }}>
                  Passwords must contain:
                </p>
                <p style={{ fontSize: '10px', margin: 0 }}>
                  * at least 1 numeric character
                </p>
                <p style={{ fontSize: '10px', margin: 0 }}>
                  * at least 1 lower case character
                </p>
                <p style={{ fontSize: '10px', margin: 0 }}>
                  * at least 1 upper case character
                </p>
                <p style={{ fontSize: '10px', margin: 0 }}>
                  * at least 1 special character
                </p>
                <p style={{ fontSize: '10px', margin: '0 0 0 1' }}>
                  * and between 8 to 15 characters
                </p>
              </div>
              <input type="password" className="form-control" id="create-password-input" placeholder="Enter password" />
            </label>
          </div>
          <div className="form-group">
            <p id="create-gender-input-label">
              Gender
            </p>
            <div className="form-check">
              <label className="form-check-label" htmlFor="flexRadioDefault1">
                Female (She/Her/Hers)
                <input className="form-check-input" type="radio" name="gender-radio" id="flexRadioDefault1" />
              </label>
            </div>
            <div className="form-check">
              <label className="form-check-label" htmlFor="flexRadioDefault2">
                Male (He/Him/His)
                <input className="form-check-input" type="radio" name="gender-radio" id="flexRadioDefault2" />
              </label>
            </div>
            <div className="form-check">
              <label className="form-check-label" htmlFor="flexRadioDefault2">
                Neutral (They/Them/Their)
                <input className="form-check-input" type="radio" name="gender-radio" id="flexRadioDefault3" />
              </label>
            </div>
          </div>
          <div className="form-group cancel-create-buttons">
            <button className="btn right-btn" type="button" onClick={() => { history.push('/login'); }}>Cancel</button>
            <button className="btn right-btn" type="button" onClick={handleSignUp}>Create</button>
          </div>
          {modalShow && (
            <Modal show={modalShow} onHide={() => setModalShow(false)}>
              <Modal.Header closeButton />
              <Modal.Body>
                <h2 style={{ 'text-align': 'center', width: '280px', height: '140px' }}>Successful Registration!</h2>
              </Modal.Body>
            </Modal>
          )}
        </form>
      </div>
    </div>
  );
}

export default Signup;
