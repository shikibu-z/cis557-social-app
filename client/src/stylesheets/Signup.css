.container {
    width: 100%;
    display: flex;
    flex-direction: column;
    gap: 10px;
}

.signupbox, .loginbox {
    gap: 20px;
    padding: 40px;
    width: 50%;
    background: rgba(196, 196, 196, 0.7);
    border: 2px solid #FFFFFF;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    border-radius: 20px;
    align-items: center;
}

.button-group {
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    align-items: right;
    gap: 10px;
    padding: 10px;
}

.btn {
    background: #F8B114;
    border-color: #F8B114;
    color: black;
    border-radius: 20px;
}

.right-btn {
    float: right;
}

.form-group {
    width: 80%;
}

.signupbox label {
    width: 100%;
}

.form-control{
    background: #FFFFFF;
    border: 1px dashed #000000;
    box-sizing: border-box;
    border-radius: 20px;
}