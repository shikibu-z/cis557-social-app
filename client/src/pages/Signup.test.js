/*eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import Signup from './Signup';

test('Profile page matches snapshot', () => {
  const component = renderer.create(<Signup />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
