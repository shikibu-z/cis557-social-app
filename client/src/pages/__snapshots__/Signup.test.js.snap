// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile page matches snapshot 1`] = `
<div>
  <div>
    <nav
      className="menu navbar navbar-expand navbar-dark bg-dark"
    >
      <div
        className="container-fluid"
      >
        <a
          className="navbar-brand"
          href="#home"
        >
          <img
            alt="connexion"
            className="d-inline-block align-top"
            id="logo"
            src="../Connexion-logos_transparent.png"
          />
           
        </a>
      </div>
    </nav>
  </div>
  <div
    className="container container d-flex align-items-center flex-column"
  >
    <img
      alt="connexion"
      id="login-box-logo"
      src="Connexion-logos_transparent.png"
    />
    <form
      className="signupbox container d-flex flex-column"
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="create-email-input"
        >
          Enter your Email address
          <input
            className="form-control"
            id="create-email-input"
            placeholder="Enter email"
            type="email"
          />
        </label>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="create-username-input"
        >
          Create a Username:
          <input
            className="form-control"
            id="create-username-input"
            placeholder="Enter username"
            type="text"
          />
        </label>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="create-password-input"
        >
          Create a Password:
          <div>
            <p
              style={
                Object {
                  "font-size": "10px",
                  "margin": 0,
                }
              }
            >
              Passwords must contain:
            </p>
            <p
              style={
                Object {
                  "fontSize": "10px",
                  "margin": 0,
                }
              }
            >
              * at least 1 numeric character
            </p>
            <p
              style={
                Object {
                  "fontSize": "10px",
                  "margin": 0,
                }
              }
            >
              * at least 1 lower case character
            </p>
            <p
              style={
                Object {
                  "fontSize": "10px",
                  "margin": 0,
                }
              }
            >
              * at least 1 upper case character
            </p>
            <p
              style={
                Object {
                  "fontSize": "10px",
                  "margin": 0,
                }
              }
            >
              * at least 1 special character
            </p>
            <p
              style={
                Object {
                  "fontSize": "10px",
                  "margin": "0 0 0 1",
                }
              }
            >
              * and between 8 to 15 characters
            </p>
          </div>
          <input
            className="form-control"
            id="create-password-input"
            placeholder="Enter password"
            type="password"
          />
        </label>
      </div>
      <div
        className="form-group"
      >
        <p
          id="create-gender-input-label"
        >
          Gender
        </p>
        <div
          className="form-check"
        >
          <label
            className="form-check-label"
            htmlFor="flexRadioDefault1"
          >
            Female (She/Her/Hers)
            <input
              className="form-check-input"
              id="flexRadioDefault1"
              name="gender-radio"
              type="radio"
            />
          </label>
        </div>
        <div
          className="form-check"
        >
          <label
            className="form-check-label"
            htmlFor="flexRadioDefault2"
          >
            Male (He/Him/His)
            <input
              className="form-check-input"
              id="flexRadioDefault2"
              name="gender-radio"
              type="radio"
            />
          </label>
        </div>
        <div
          className="form-check"
        >
          <label
            className="form-check-label"
            htmlFor="flexRadioDefault2"
          >
            Neutral (They/Them/Their)
            <input
              className="form-check-input"
              id="flexRadioDefault3"
              name="gender-radio"
              type="radio"
            />
          </label>
        </div>
      </div>
      <div
        className="form-group cancel-create-buttons"
      >
        <button
          className="btn right-btn"
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
        <button
          className="btn right-btn"
          onClick={[Function]}
          type="button"
        >
          Create
        </button>
      </div>
    </form>
  </div>
</div>
`;
